### ISSUE TITLE
Performance Optimizations Tracker

### ISSUE THREAD

**Issue by dapplion** [ChainSafe/lodestar#3466]
# 📋  Milestone/Epic Tracker    ![GitHub issues by-label](https://img.shields.io/github/issues/chainsafe/lodestar/scope-performance)
This tracker is to monitor outstanding issues and PRs relating to performance optimizations for Lodestar:

### 🟥  High Priority
- [ ] I/O lag issue in under-resourced machine https://github.com/ChainSafe/lodestar/issues/4002
- [ ] https://github.com/ChainSafe/lodestar/issues/3465
   - Impact: 5% total CPU
   - Idea: Use native **gossip crypto**
- [ ] https://github.com/ChainSafe/lodestar/issues/2457
   - Impact: 10% total CPU
   - Idea: Faster **gossip deserialization**
- [ ] https://github.com/ChainSafe/lodestar/issues/3080
   - Impact: 90% of processBlock
   - Idea: Optimize **processAttestations**
- [ ] https://github.com/ChainSafe/lodestar/issues/3116
   - Idea: Reduce **export** cost
- [ ] https://github.com/ChainSafe/lodestar/issues/3000
   - Impact: 5% of total CPU
   - `protons` lib performance issue
- [ ] https://github.com/ChainSafe/lodestar/issues/2488
- [ ] https://github.com/ChainSafe/lodestar/issues/2154

### 🟧  Medium Priority
- [ ] https://github.com/ChainSafe/lodestar/issues/3402
   - Idea: Optimize att gossip validation
- [ ] https://github.com/ChainSafe/lodestar/issues/2425
   - Idea: **Initial sync** verify sigs in parallel
- [ ] https://github.com/ChainSafe/lodestar/issues/3415
   - Idea: Only **TreeBacked** states
- [ ] https://github.com/ChainSafe/lodestar/issues/2846
   - Idea: Optimize **regen** cost
- [ ] https://github.com/ChainSafe/lodestar/issues/3411
   - Idea: Optimize **logger** formatting
- [ ] https://github.com/ChainSafe/lodestar/issues/3098
   - Idea: Call produceAttestationData once
- [ ] https://github.com/ChainSafe/lodestar/issues/3082
   - Idea: Optimize **processEth1Data**
- [ ] https://github.com/ChainSafe/lodestar/issues/3443

### 🟨  Low Priority
- [ ] https://github.com/ChainSafe/lodestar/issues/2877
   - Idea: Optimize att gossip validation
- [ ] https://github.com/ChainSafe/lodestar/issues/3439
   - Idea: Batch level **db** ops
- [ ] https://github.com/ChainSafe/lodestar/issues/2749
   - Idea: Use raw **for loops**
- [ ] https://github.com/ChainSafe/lodestar/issues/3414
   - Idea: Cache indexed attestations
- [ ] https://github.com/ChainSafe/lodestar/issues/2747
   - Impact: 1% of total CPU
   - Idea: Improve **forkchoice** updateHead
- [ ] https://github.com/ChainSafe/lodestar/issues/2857
   - Idea: Sync hashBlocks performance issue
- [ ] https://github.com/ChainSafe/lodestar/issues/2527
   - Idea: Cheaper RAND seeding
- [ ] https://github.com/ChainSafe/lodestar/issues/2352
   - Idea: **unshuffleList** performance
- [ ] https://github.com/ChainSafe/lodestar/issues/2206
   - Idea: Improve **hashing** performance
- [ ] https://github.com/ChainSafe/lodestar/issues/2050
   - Idea: Minimize Garbage Collector **GC** cost

**Comment by philknows** [comment]
Not a scalable tracking solution for issues. Closing.
